### PR TITLE
Compatibility gazebo 4.0

### PR DESCRIPTION
**Pull request by hsu**
API changes moving from Gazebo 2.0 to 4.0

### CODE REVIEW

**Comment by adolfo-rt** [comment]
`gazebo_ros_control` part LGTM. Thanks for updating the doc, and not only the code.

File: gazebo_plugins/CMakeLists.txt [hunk 1]
```diff
@@ -38,7 +38,7 @@ endif()
 
 # Depend on system install of Gazebo and SDFormat
 find_package(gazebo REQUIRED)
-find_package(PCL REQUIRED)
+# find_package(PCL REQUIRED)
```
Comment (scpeters) [inline gazebo_plugins/CMakeLists.txt:41]: I would just delete this line

File: gazebo_ros/CMakeLists.txt [hunk 2]
```diff
@@ -56,6 +56,10 @@ catkin_package(
 
 include_directories(${Boost_INCLUDE_DIRS} ${catkin_INCLUDE_DIRS} ${GAZEBO_INCLUDE_DIRS} ${SDFormat_INCLUDE_DIRS} ${TinyXML_INCLUDE_DIRS})
 
+link_directories(
+  # ${GAZEBO_LIBRARY_DIRS}
```
Comment (scpeters) [inline gazebo_ros/CMakeLists.txt:60]: delete this commented line?

**Comment by scpeters** [comment]
Sorry I made some conflicts by merging #315

**Comment by j-rivero** [comment]
This looks out of date to me. Is this still needed or we can close it?

**Comment by j-rivero** [comment]
please reopen if needed.

**Pull request closed without merging**